lib/mixins/_writable.js

```diff
@@ -49,27 +59,40 @@ _Writable.prototype.firstFinish = function(handler, ignoreCloseAsFinish) {
 		// (not in addition to) 'finish'
 		this.once('close', handleEvent);
 	}
+
+	if(!!this._writableState.zEnded) {
```

Comment:
You're handling this here in firstFinish() but not in firstUnignoredError() .  Be consistent.